### PR TITLE
Add Costume type to model

### PR DESCRIPTION
**Pull request by NickCondron**
Color names based on [ssbwiki.com](https://www.ssbwiki.com/Alternate_costume_(SSBM)), but I made sure Red, Green, and Blue existed for every character. ~I plan to add colloquial names if #8 gets merged.~

I removed derive implementations of Default/Deserialize for the Player type because my change caused conflicts. It doesn't seem like those traits made much sense in the first place.

~It might make sense to move the `default`, `red`, `green`, `blue` functions to the External type, but I left it how it is for now.~

### CODE REVIEW

**Comment by hohav** [comment]
Looks great. Thanks!

How do you anticipate `red()` & co being used? I assume for teams, but to do what exactly?

**Comment by NickCondron** [comment]
I honestly didn't have a solid use case in mind. I was having fun with macros and it seemed like a simple addition. I understand wanting to keep the api streamlined, so I'm not sentimental about those functions.

Maybe a niche use case would be searching for games with red characters but you don't care which character in particular.

**Comment by hohav** [comment]
> I honestly didn't have a solid use case in mind. I was having fun with macros and it seemed like a simple addition. I understand wanting to keep the api streamlined, so I'm not sentimental about those functions.

I'd rather leave them out for now, in that case. We can always add them if there's demand, but removing them would break backwards compatibility.

**Comment by NickCondron** [comment]
Ok I removed them for now. I plan on using this costume feature. If I find that the red/green/blue functions would be useful for me I will bug you to add them back in at that point.

File: peppi/src/model/enums/costume.rs [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+		impl std::convert::TryFrom<(&str, External)> for $name {
+			type Error = crate::model::pseudo_enum::ConversionError;
+			fn try_from(i: (&str, External)) -> std::result::Result<Self, Self::Error> {
+				match i.1 {
+					$( External::$external => Ok($name::$variant($variant_type::try_from(i.0)?)), )*
+					_ => Ok($name::$unknown(0)),
+				}
+			}
+		}
```
[the remark this answers is not in the document]
Comment (NickCondron) [reply peppi/src/model/enums/costume.rs]: Yea I added this for consistency with pseudo_enum/action_state. I agree it doesn't really make sense to support this. Removed.

**Comment by NickCondron** [comment]
Rebased and squashed

**Review by hohav — COMMENTED** [review]
One last thing: could you replace spaces with tabs? Sorry I haven't gotten around to configuring rustfmt yet.

**Comment by NickCondron** [comment]
> One last thing: could you replace spaces with tabs? Sorry I haven't gotten around to configuring rustfmt yet.

I'm not seeing any space indents on this branch

**Comment by hohav** [comment]
> > One last thing: could you replace spaces with tabs? Sorry I haven't gotten around to configuring rustfmt yet.
> 
> I'm not seeing any space indents on this branch

Oh my bad, Firefox was misleading me. Sorry for the noise.

**Pull request merged**